### PR TITLE
Addressed 2 issues + added a section to an effected guide.

### PR DESCRIPTION
**Pull request by cklyne**
I addressed the following issues:
 - [#42446](https://github.com/tensorflow/tensorflow/issues/42446) in the tensorflow repo.
 - #802 in the keras-io repo.

I also added to the `training_with_built_in_methods` guide a demonstration of implementing a custom loss class using `LossFunctionWrapper`. I'm not sure if this is best practice since this class is not exported with `@keras_export`, but I believe it would be helpful for people who need this guide.

This is my first time contributing to an open source project, so I would appreciate comments and criticism :)

P.S. should I update the `Last modified` section in the 2 effected guides?

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thanks for the PR!

**Comment by cklyne** [comment]
> LGTM, thanks for the PR!

Cool, should I add the .md & .ipynb files now?
And should I update the date in the `Last modified` section of the tutorials?

**Comment by fchollet** [comment]
> Cool, should I add the .md & .ipynb files now?

Please do! You can just run the `add_guide` command.

> And should I update the date in the Last modified section of the tutorials?

Sure!

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
